products/BellHybrid/services/desktop/endpoints/deviceInfo/DeviceInfoEndpoint.cpp

```diff
@@ -23,13 +24,98 @@ namespace sdesktop::endpoints
     {
         switch (context.getMethod()) {
         case http::Method::get:
-            getDeviceInfo(context);
+            handleGet(context);
             break;
         default:
             context.setResponseStatus(http::Code::BadRequest);
-            sender::putToSendQueue(context.createSimpleResponse());
             break;
         }
+        sender::putToSendQueue(context.createSimpleResponse());
+    }
+
+    auto DeviceInfoEndpoint::handleGet(Context &context) -> void
+    {
+        const auto &requestBody = context.getBody();
+
+        if (requestBody[json::fileList].is_number()) {
+
+            const auto diagFileType = static_cast<DiagnosticFileType>(requestBody[json::fileList].int_value());
+
+            if (!magic_enum::enum_contains<DiagnosticFileType>(diagFileType)) {
+                LOG_ERROR("Bad diagnostic type %d requested", static_cast<unsigned>(diagFileType));
```

Comment:
why not (returning enum):
``` 
const auto diagFileType = parseDiagnosticFileType(requestBody[jeson::fileList]);
```
then just use the diagFileType in log.
Especially that you can just use a magic enum to print string evaluation of that enum here. But there would be cast on cast here then.

This way we would have:
- separate parsing logic from switch case handling
- one place where casting is done instead  is in analysis if it's in, then cast )
- more self explaining, readable code :)